R/txrange.R

```diff
@@ -0,0 +1,408 @@
+suppressMessages(require(stringr))
+library(BSgenome)
+library(glmnet)
+library(xgboost)
+
+txrange.scoreReadClasses = function(se, genomeSequence, annotations, 
+  withAdapters = FALSE, min.readCount = 2){
+    options(scipen = 999)
+    se = addRowData(se, genomeSequence, annotations)
+    thresholdIndex = which(rowSums(assays(se)$counts)
+        >=min.readCount)
+    rowData(se)$geneScore = getGeneScore(se, thresholdIndex, 
+      method = "xgboost")
+    rowData(se)$txScore = getTranscriptScore(se, thresholdIndex, 
+      method = "xgboost")
+
+    return(se)
+}
+
+addRowData = function(se, genomeSequence, annotations){
+  exons = str_split(rowData(se)$intronStarts,",")
+  rowData(se)$numExons = sapply(exons, FUN = length)+1
+  rowData(se)$numExons[is.na(exons)] = 1
+  classifications = getReadClassClassifications(rowRanges(se), annotations)
+  rowData(se)$equal = classifications$equal
+  rowData(se)$compatible = classifications$compatible
+  rowData(se)$compatibleCount = classifications$compatibleCount
+  rowData(se)$GENEID = assignGeneIds(rowRanges(se), annotations)
+  rowData(se)$novel = grepl("gene.", 
+      rowData(se)$GENEID)
+  se = calculateGeneProportion(se)
+  se = countPolyATerminals(se, genomeSequence)
+ 
+  return(se)
+}
+
+calculateGeneProportion = function(resultOutput){
+  countsTBL <- as_tibble(assays(resultOutput)$counts) %>%
+    mutate(geneId = rowData(resultOutput)$GENEID) %>%
+    group_by(geneId) %>%
+    mutate_at(vars(-geneId), .funs = sum) %>%
+    ungroup() %>%
+    dplyr::select(-geneId)
+  geneReadProp <- assays(resultOutput)$counts / countsTBL
+  assays(resultOutput, withDimnames = F)$geneReadProp = geneReadProp
+  rowData(resultOutput)$totalGeneReadProp = 
+    rowSums(assays(resultOutput)$counts / rowSums(countsTBL))
+  return(resultOutput)
+}
+
+convertSEtoGRangesList = function(combinedOutputs){
```

Comment:
is this function required/used?

Reply:
Not anymore, was made redundant in the last commit. I will remove this. 352bc2a